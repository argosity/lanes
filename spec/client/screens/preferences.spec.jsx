import { React, Snapshot, getScreenInstance } from 'hippo/testing/index'; // eslint-disable-line
import Preferences from 'hippo/screens/preferences';

const screenInstance = getScreenInstance('preferences');

describe('Screen Preferences', () => {
    it('renders and matches snapshot', () => {
        expect(Snapshot(<Preferences screen={screenInstance} />)).toMatchSnapshot();
    });
});
