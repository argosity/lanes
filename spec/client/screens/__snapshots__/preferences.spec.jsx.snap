// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Preferences renders and matches snapshot 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #DDDDDD;
  color: #333333;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  grid-area: header;
  margin: undefined;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 12px;
  padding-bottom: 12px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 100%;
  min-width: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 12px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 100%;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  display: grid;
  height: 100%;
  grid-template-areas: "header header" "main" "footer footer";
  grid-template-columns: 100%;
  grid-gap: 12px 12px;
  grid-template-rows: 48px 1fr 96px;
}

.c8 {
  font-size: 24px;
  line-height: 1.333;
  max-width: 576px;
  font-weight: 300;
  margin-top: 0;
  margin-bottom: 0;
}

.c17 {
  padding: 11px;
  border: 1px solid rgba(0,0,0,0.15);
  border-radius: 4px;
  outline: none;
  background-color: transparent;
  color: inherit;
  font: inherit;
  margin: 0;
  border: none;
  width: 100%;
  -webkit-appearance: none;
}

.c17::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c17::-moz-placeholder {
  color: #AAAAAA;
}

.c17:-ms-input-placeholder {
  color: #AAAAAA;
}

.c17::-moz-focus-inner {
  border: none;
  outline: none;
}

.c16 {
  width: 100%;
}

.c13 {
  font-size: 16px;
  line-height: 1.375;
}

.c14 {
  font-size: 16px;
  line-height: 1.375;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #777777;
}

.c11 {
  height: 100%;
  min-width: 0;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: white no-repeat;
  background-image: linear-gradient(to bottom,#00CCEB,#00CCEB),linear-gradient(to bottom,rgba(0,0,0,0.15),rgba(0,0,0,0.15));
  background-size: 0 2px,100% 1px;
  background-position: 50% 100%,50% 100%;
  -webkit-transition: background-size 0.3s cubic-bezier(0.64,0.09,0.08,1);
  transition: background-size 0.3s cubic-bezier(0.64,0.09,0.08,1);
}

.c11.focused,
.c11:focus-within {
  background-size: 100% 2px,100% 1px;
  outline: none;
}

.c11.focused span,
.c11:focus-within span {
  font-weight: 500;
}

.c11 input[type="checkbox"] {
  margin-left: 12px;
}

.c18 {
  height: 100%;
  min-width: 0;
  -webkit-align-content: space-around;
  -ms-flex-line-pack: space-around;
  align-content: space-around;
  grid-column-end: span 1;
  grid-row-end: span 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: white no-repeat;
  background-image: linear-gradient(to bottom,#00CCEB,#00CCEB),linear-gradient(to bottom,rgba(0,0,0,0.15),rgba(0,0,0,0.15));
  background-size: 0 2px,100% 1px;
  background-position: 50% 100%,50% 100%;
  -webkit-transition: background-size 0.3s cubic-bezier(0.64,0.09,0.08,1);
  transition: background-size 0.3s cubic-bezier(0.64,0.09,0.08,1);
}

.c18.focused,
.c18:focus-within {
  background-size: 100% 2px,100% 1px;
  outline: none;
}

.c18.focused span,
.c18:focus-within span {
  font-weight: 500;
}

.c18 input[type="checkbox"] {
  margin-left: 12px;
}

.c10 {
  display: grid;
  grid-auto-flow: row dense;
  grid-auto-rows: minmax(20px,auto);
  grid-template-columns: repeat(auto-fill,minmax(120px,1fr));
  grid-gap: 8px 8px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-overflow-scrolling: touch;
  padding: 12px;
  grid-auto-rows: minmax(80px,auto);
}

.c10 > * {
  min-height: 85px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c9 {
  grid-area: body;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c0 * {
  box-sizing: inherit;
}

.c3 {
  cursor: pointer;
  outline: none;
  font: inherit;
  text-decoration: none;
  font: inherit;
  margin: 0;
  background-color: transparent;
  overflow: visible;
  text-transform: none;
  text-align: center;
  display: inline-block;
  min-width: 96px;
  max-width: 384px;
  font-weight: 600;
  background-color: #865CD6;
  color: rgba(255,255,255,0.85);
  border: none;
  font-size: 1.1875rem;
  line-height: 24px;
  padding: 10px 10px;
}

.c3 svg {
  fill: #FFFFFF;
  stroke: #FFFFFF;
  -webkit-transition: none;
  transition: none;
}

.c3:hover {
  box-shadow: 0px 0px 0px 2px #865CD6;
}

.c4 {
  display: inline-block;
}

.c4:first-child:not(:last-child) {
  margin-right: 12px;
}

.c4 > * {
  vertical-align: bottom;
}

.c5 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  fill: #666666;
  stroke: #666666;
}

.c5 g {
  fill: inherit;
  stroke: inherit;
}

.c5 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c5 *[stroke*="#"],
.c5 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c5 *[fill-rule],
.c5 *[FILL-RULE],
.c5 *[fill*="#"],
.c5 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c6:first-child:not(:last-child) {
  margin-right: 12px;
}

@media (min-width:992px) {
  .c11 {
    grid-column-end: span 2;
  }
}

@media (min-width:768px) {
  .c11 {
    grid-column-end: span 2;
  }
}

@media (min-width:300px) {
  .c11 {
    grid-column-end: span 2;
  }
}

@media (min-width:992px) {
  .c18 {
    grid-column-end: span 3;
  }
}

@media (min-width:768px) {
  .c18 {
    grid-column-end: span 3;
  }
}

@media (min-width:300px) {
  .c18 {
    grid-column-end: span 3;
  }
}

@media only screen and (min-width:700px) {
  .c3 {
    -webkit-transition: 0.1s ease-in-out;
    transition: 0.1s ease-in-out;
  }
}

<div
  className="c0"
>
  <div
    className="hippo-screen preferences is-active c1"
    data-screen-id="preferences"
    rows={
      Array [
        "xxsmall",
        "flex",
        "xsmall",
      ]
    }
  >
    <div
      aria-label={undefined}
      className="hippo-toolbar c2"
      direction="row"
      fill="horizontal"
    >
      <button
        aria-label={undefined}
        className="save-button c3"
        disabled={false}
        href={undefined}
        icon={<Save />}
        label="Save"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          aria-hidden={true}
          className="c4"
        >
          <svg
            aria-label="Save"
            className="c5"
            height="24px"
            role="img"
            version="1.1"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M3,2 L3,21 L21,21 L21,3 L12,3 L12,14 M8,11 L12,15 L16,11"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </span>
        <span
          className="c6"
        >
          Save
        </span>
      </button>
    </div>
    <div
      aria-label={undefined}
      className="c7"
      direction="column"
    >
      <h3
        className="c8"
      >
        Account information
      </h3>
    </div>
    <div
      className="c9"
    >
      <div
        className="c10"
        columns="repeat(auto-fill, minmax(120px, 1fr))"
        flow="row dense"
      >
        <label
          className="form-field-wrapper c11"
          height={undefined}
          type="text"
          width={undefined}
        >
          <div
            aria-label={undefined}
            className="labels c12"
            direction="row"
            wrap={true}
          >
            <span
              className="c13"
            >
              Login
            </span>
            <span
              className="c14"
              color="dark-5"
            />
          </div>
          <div
            className="c15"
          >
            <div
              className="c16"
            >
              <input
                autoComplete="off"
                autoFocus={undefined}
                className="c17"
                defaultValue={undefined}
                id={undefined}
                name="login"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
        </label>
        <label
          className="form-field-wrapper c18"
          height={undefined}
          type="text"
          width={undefined}
        >
          <div
            aria-label={undefined}
            className="labels c12"
            direction="row"
            wrap={true}
          >
            <span
              className="c13"
            >
              Name
            </span>
            <span
              className="c14"
              color="dark-5"
            />
          </div>
          <div
            className="c15"
          >
            <div
              className="c16"
            >
              <input
                autoComplete="off"
                autoFocus={undefined}
                className="c17"
                defaultValue={undefined}
                id={undefined}
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
        </label>
        <label
          className="form-field-wrapper c18"
          height={undefined}
          type="text"
          width={undefined}
        >
          <div
            aria-label={undefined}
            className="labels c12"
            direction="row"
            wrap={true}
          >
            <span
              className="c13"
            >
              Email
            </span>
            <span
              className="c14"
              color="dark-5"
            />
          </div>
          <div
            className="c15"
          >
            <div
              className="c16"
            >
              <input
                autoComplete="off"
                autoFocus={undefined}
                className="c17"
                defaultValue={undefined}
                id={undefined}
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
        </label>
        <label
          className="form-field-wrapper c11"
          height={undefined}
          type="password"
          width={undefined}
        >
          <div
            aria-label={undefined}
            className="labels c12"
            direction="row"
            wrap={true}
          >
            <span
              className="c13"
            >
              Password
            </span>
            <span
              className="c14"
              color="dark-5"
            />
          </div>
          <div
            className="c15"
          >
            <div
              className="c16"
            >
              <input
                autoComplete="off"
                autoFocus={undefined}
                className="c17"
                defaultValue={undefined}
                id={undefined}
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                tabIndex={0}
                type="password"
                value=""
              />
            </div>
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;
